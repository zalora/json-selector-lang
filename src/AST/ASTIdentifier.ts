// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.

import { Token } from '../tokens';
import ASTExpression from './ASTExpression';

class ASTIdentifier implements ASTExpression {
  token: Token;
  value: string;

  constructor(token: Token, value: string) {
    this.token = token;
    this.value = value;
  }

  toString(): string {
    return this.value;
  }

  tokenLiteral(): string {
    return this.token.literal;
  }
}

export default ASTIdentifier;
