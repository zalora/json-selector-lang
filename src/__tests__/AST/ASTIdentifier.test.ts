// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.

import ASTIdentifier from '../../AST/ASTIdentifier';
import { ident } from '../../tokens';

describe('ASTIdentifier', () => {
  const token = { type: ident, literal: 'data' };

  it('tests instance with token input', () => {
    const astIdent = new ASTIdentifier(token, token.literal);
    expect(astIdent).toMatchSnapshot();
  });

  it('tests toString returns value of the identifier', () => {
    const astIdent = new ASTIdentifier(token, token.literal);
    expect(astIdent.toString()).toMatchSnapshot();
  });

  it('tests tokenLiteral returns literal of the token', () => {
    const astIdent = new ASTIdentifier(token, token.literal);
    expect(astIdent.tokenLiteral()).toMatchSnapshot();
  });
});
