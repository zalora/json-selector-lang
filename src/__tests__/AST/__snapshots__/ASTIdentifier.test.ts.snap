// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASTIdentifier tests instance with token input 1`] = `
ASTIdentifier {
  "token": Object {
    "literal": "data",
    "type": Symbol(ident),
  },
  "value": "data",
}
`;

exports[`ASTIdentifier tests toString returns value of the identifier 1`] = `"data"`;

exports[`ASTIdentifier tests tokenLiteral returns literal of the token 1`] = `"data"`;
